Restructure Docs (#18)
CodeMC/Documentation: docs/usage-guidelines.md

# CodeMC Usage Guidelines
By joining CodeMC are you agreeing to the below written Usage Guidelines ("Guidelines"). Not following them will result in punishments, as described within the [Punishments](#punishments) section of this page.

## Project Standards
Projects on the CodeMC CI need to follow certain standards to be allowed to stay on it.

### Project must be of good quality
The Code of your project should display a high level of quality and effort.  
This includes, but isn't limited to following Coding Practices such as **D**on't **R**epeat **Y**ourself (DRY), SOLID, etc., properly documenting code to explain more complicated parts and similar.

### Project must be of use for others
While it can be nice to have a CI for your library to upload and share, should you limit it to libraries and projects that can have a use for others.  

> Effectively, if the project's purpose is to ease your coding of other Projects, while not having any use outside of that for others, is it not something you can and should put on our CI.

Reviewer: ```suggestion
Effectively, if the project's purpose is to ease your coding of other Projects, while not having any use outside of that for others, is it something you shouldn't put on our CI.
```

Do note that we consider the following as maintaining a project:

- Creating commits with proper messages.
- Keeping dependencies updated.
- Keeping the project's issue and Pull request tracker clean by closing issues, closing/merging pull requests, labeling them, assigning people, etc.

With all this said are we aware that there can be justifiable reasons for a projects being stale. In such cases do we ask you to inform us about it.

### Project cannot encourage nor support illegal activities or content
We want the programming community to be as welcoming and inclusive as possible. As such are we not tollerating projects that encourage, include or support hate-speech, racism, sexism, bigotry or any other form of harassment, or illegal content.  
Should you find such a project on our CI are your obligated to report it to us as soon as possible to be dealt with.

### Project should have an english documentation available
CodeMC accepts non-english projects. However, to allow a user to easily understand what this project does and is about, should it provide easy access to an english documentation of the Project's README or other noteworthy documentation a user may want to read.

> Translations done through a translator without proofreading by an english speaking person should have a note of not being 100% accurate and done via translators.

Reviewer: ```suggestion
Translations done through a translator without proofreading by an English-speaking person should note less than 100% accuracy, indicating the use of a translator.
```

## Reasonable Usage Limits
We do not apply usage limits (Rate limits) on most of our CI. For parts where we do this are we giving relatively generous limits.  
In order to keep this state and not force harsher limits on anyone are we requesting you to not abuse these limits in any way. This means to keep frequency of builds and uploads to the Nexus to a reasonable limit.

## Punishments
We reserve the right to punish you in the following ways, should you be confirmed to have broken these guidelines:

- Restricting access to our Nexus and CI.
- Removal of builds and Nexus uploads.
- Removal of your Project(s).
- Removal of your account(s).
- Complete blocking of access to our CI and/or Nexus.

> CodeMC may issue one or multiple of the above punishments at any given time at our own discretion.

Reviewer: I usually like to keep things explicit and clear whenever punishments are involved.
```suggestion
CodeMC may issue one or multiple of the above punishments at any given time at our discretion. Any of these actions may happen in any order of any severity. 
```

## Guideline Updates and Opting out

> These guidelines may be updated at any given moment. We may notify users on our [Discord Server](https://discord.gg/AGcFMu6) about updates to these Guidelines, do not guarantee this however.  

Reviewer: Updating rules without proper notifications sounds lazy. I would explicitly state that notifications are for updates that matter.
```suggestion
These guidelines may be updated at any given moment. We will notify users on our [Discord Server](https://discord.gg/AGcFMu6) about significant updates to these Guidelines.
```

Author: I wouldn't guarantee this, especially since it would make it a difficult task, as we would now need to decide on what is a "significant" change worth notifying people.
I have my doubts we will actually make many updates to these guidelines anyways, so I'll leave this as it is right now.

Reviewer: Gotcha. I think "significant" could be defined as an impact that changes the way we would usually expect people to interact with the service (e.g. we now require 2FA on GitHub accounts)

## Reasonable Usage Limits

We do not apply usage limits (Rate limits) on most of our Services. For parts where we do this are we giving relatively generous limits.  
In order to keep this state and not force harsher limits on anyone are we requesting you to not abuse these limits in any way. This means to keep frequency of builds and uploads to the Nexus to a reasonable limit.

## Punishments

We reserve the right to punish you in the following ways, should you be confirmed to have broken these guidelines:

- Restricting access to our Nexus and CI.
- Removal of builds and Nexus uploads.
- Removal of your Project(s).
- Removal of your account(s).
- Partial or complete blocking of access to our Services.

CodeMC may issue one or multiple of the above punishments at any given time at our discretion. Any of these actions may happen in any order of any severity. 

## Guideline Updates and Opting out

These guidelines may be updated at any given moment and we will inform about those on our [Discord Server](https://discord.gg/AGcFMu6) unless not possible for technical reasons.  

> Should we update these Guidelines will we grant a 1 Wekk (7 Days) Period to Opt out of these Guidelines. You may Opt out by no longer using our Services. By continuing to using our services after the aforementioned time Period are you agreeing to the updated Guidelines.

Reviewer: Missed typos.
```suggestion
Should we update these Guidelines will we grant a 1 Week (7 Days) Period to Opt out of these Guidelines. You may Opt-out by no longer using our Services. By continuing to use our services after the aforementioned period are you agreeing to the updated Guidelines.
```
